extension-manifest-v3/scripts/build.js

```diff
@@ -57,13 +57,14 @@ if (argv.target.startsWith('safari')) {
   argv.target = 'safari';
 }
 
+// Add flags to manifest
+if (argv.debug) manifest.debug = true;
+if (argv.staging) manifest.staging = true;
```

Comment:
was wondering why you "invent" the stating mode, but from the other changes it looks like it was implicitly there for some subsystems.

We have to decide how those modes should work. 

Historically:
* debug - was setting all backends to staging and was enabling some additional logging
* staging - was recently introduced to download adblocker engines from the staging environment

It looks like we have 3 separate aspects to deal with:
* debugging of the addon (logs, some settings like opera serp, scritplets debugging)
* adblocker engines environment
* backends (staging vs production)

Reply:
The staging mode usually does not work. They it yourself. The engines are not updated there (I suppose we have to trigger it manually). 

Firstly, there was only a debug mode, but with that issue I had to add separate flag for loading engines from staging.

Reply:
ah, so you mean even if we downloaded staging engine pre-build, they would update to production engine in the runtime anyways?

I'm fine to keep the staging mode. Shall we then update all endpoints configuration to use stagingMode instead of debugMode?

Reply:
`npm start -- --staging` fails... as the staging CDN does not have the proper version of the engines..

For me `--staging` should only set the engines' endpoints - as this must be used only when we know that we pushed something to the staging CDN for engines and only engines.

Reply:
the problem occurs as we don't create staging engines on every adblocker release - perhaps we should. When this happen please reach out so we can schedule a CI build.